packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -2091,23 +2091,32 @@ class _InactiveElements {
 
   static void _deactivateRecursively(Element element) {
     assert(element._lifecycleState == _ElementLifecycle.active);
-    element.deactivate();
-    assert(element._lifecycleState == _ElementLifecycle.inactive);
-    element.visitChildren(_deactivateRecursively);
-    assert(() {
-      element.debugDeactivated();
-      return true;
-    }());
+    try {
+      element.deactivate();
+    } catch (_) {
+      element._ensureDeactivated();
```

Comment:
This may potentially be called twice if the crash is in `_ensureDeactivated`

Reply:
don't think this is an issue though, but may make stack trace a bit messy

Reply:
I tried removing the "ensure" part out of `deactivate` but one of our tests is calling `deactivate` directly so I couldnt. See https://github.com/flutter/flutter/pull/173148#discussion_r2249048599

Reply:
yeah I think this is fine